### PR TITLE
Add REST API to download trial log

### PR DESCRIPTION
**Pull request by liuzhe-lz**

### CODE REVIEW

File: src/nni_manager/training_service/pai/paiTrainingService.ts [hunk 1]
```diff
@@ -130,6 +130,20 @@ class PAITrainingService implements TrainingService {
         return Promise.resolve(paiTrialJob);
     }
 
+    public async getTrialLog(trialJobId: string, logType: LogType): Promise<string> {
+        const dir: string = HDFSClientUtility.getHdfsTrialWorkDir(this.paiClusterConfig!.userName, trialJobId);
```
Comment (chicm-ms) [inline src/nni_manager/training_service/pai/paiTrainingService.ts]: The log files does not exist when trial job is running for PAI,  should we check job status here?

File: src/nni_manager/common/trainingService.ts [hunk 2]
```diff
@@ -24,6 +24,8 @@
  */
 type TrialJobStatus = 'UNKNOWN' | 'WAITING' | 'RUNNING' | 'SUCCEEDED' | 'FAILED' | 'USER_CANCELED' | 'SYS_CANCELED' | 'EARLY_STOPPED';
 
+type LogType = 'TRIAL_LOG' | 'TRIAL_STDERR';
```
Comment (SparkSnail) [inline src/nni_manager/common/trainingService.ts]: `TRIAL_STDOUT` ?

File: src/nni_manager/training_service/pai/paiTrainingService.ts [hunk 3]
```diff
@@ -130,6 +130,20 @@ class PAITrainingService implements TrainingService {
...
+        let logPath: string;
+        if (logType === 'TRIAL_LOG') {
+            logPath = unixPathJoin(dir, 'trial.log');
+        } else if (logType === 'TRIAL_STDERR') {
```
Comment (SparkSnail) [inline src/nni_manager/training_service/pai/paiTrainingService.ts]: There is no `stderr` file in pai mode.

File: src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts [hunk 4]
```diff
@@ -205,6 +205,19 @@ class RemoteMachineTrainingService implements TrainingService {
         }
     }
 
+    public getTrialLog(trialJobId: string, logType: LogType): Promise<string> {
+        const client: Client = this.trialSSHClientMap.get(trialJobId)!;
+        let logPath: string;
+        if (logType === 'TRIAL_LOG') {
+            logPath = unixPathJoin(this.remoteExpRootDir, 'trials', trialJobId, 'trial.log');
+        } else if (logType === 'TRIAL_STDERR') {
```
Comment (SparkSnail) [inline src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts]: there is no `stderr` file in remote mode.

**Comment by liuzhe-lz** [comment]
Seemingly remains a lot things to do.
Suggest defer this PR to next release.

**Pull request closed without merging**